rust/nft-candy-machine-v2/src/lib.rs

```diff
@@ -0,0 +1,1003 @@
+            ctx.accounts.rent.to_account_info(),
+            candy_machine_creator.to_account_info(),
+        ];
+        msg!("Before metadata");
+        sol_log_compute_units();
+
+        invoke_signed(
+            &create_metadata_accounts(
+                *ctx.accounts.token_metadata_program.key,
+                *ctx.accounts.metadata.key,
+                *ctx.accounts.mint.key,
+                *ctx.accounts.mint_authority.key,
+                *ctx.accounts.payer.key,
+                candy_machine_creator.key(),
+                config_line.name,
+                candy_machine.data.symbol.clone(),
+                config_line.uri,
+                Some(creators),
+                candy_machine.data.seller_fee_basis_points,
+                true,
+                candy_machine.data.is_mutable,
+            ),
+            metadata_infos.as_slice(),
+            &[&authority_seeds],
+        )?;
+
+        msg!("Before master");
+        sol_log_compute_units();
+        invoke_signed(
+            &create_master_edition(
+                *ctx.accounts.token_metadata_program.key,
+                *ctx.accounts.master_edition.key,
+                *ctx.accounts.mint.key,
+                candy_machine_creator.key(),
+                *ctx.accounts.mint_authority.key,
+                *ctx.accounts.metadata.key,
+                *ctx.accounts.payer.key,
+                Some(candy_machine.data.max_supply),
+            ),
+            master_edition_infos.as_slice(),
+            &[&authority_seeds],
+        )?;
+
+        let mut new_update_authority = Some(candy_machine.authority);
+
+        if !candy_machine.data.retain_authority {
+            new_update_authority = Some(ctx.accounts.update_authority.key());
+        }
+
+        msg!("Before update");
+        sol_log_compute_units();
+        invoke_signed(
+            &update_metadata_accounts(
+                *ctx.accounts.token_metadata_program.key,
+                *ctx.accounts.metadata.key,
+                candy_machine_creator.key(),
+                new_update_authority,
+                None,
+                Some(true),
+            ),
+            &[
+                ctx.accounts.token_metadata_program.to_account_info(),
+                ctx.accounts.metadata.to_account_info(),
+                candy_machine_creator.to_account_info(),
+            ],
+            &[&authority_seeds],
+        )?;
+
+        msg!("Before instr check");
+        sol_log_compute_units();
+
+        let instruction_sysvar_account_info = instruction_sysvar_account.to_account_info();
+
+        let instruction_sysvar = instruction_sysvar_account_info.data.borrow();
+        let current_instruction = sysvar::instructions::load_current_index(&instruction_sysvar);
+        let mut idx = 0;
+        let num_instructions = read_u16(&mut idx, &instruction_sysvar)
+            .map_err(|_| ProgramError::InvalidAccountData)?;
+
+        if current_instruction < num_instructions - 1 {
```

Comment:
if this is meant to defeat the issue with multiple mints per tx, then this works I think.

if this is meant to defeat cherry pick ix then I think it doesn't matter if cherry ix comes before cm mint ix -- maybe not as important now with captcha and post reveal features though.